src/core/include/hydrogen/Preferences.h

```diff
@@ -1293,6 +1317,13 @@ inline QString Preferences::getH2ProcessName() {
 	return m_sH2ProcessName;
 }
 
+inline QString Preferences::getPreferencesOverwritePath() {
```

Comment:
I'm wondering if this should go into our `Filesystem` class, since this already manages other things like the temporary directory or the user/system dir. 

But it is great that you've integrated this option already, this will come handy for the configuration for the hydrogen LV2 plugin!

Reply:
Sure. Moving it to `Filesystem` will keep the `Preferences` class more focused.

The reason I didn't put functions like `NsmClient::copyPreferences` and `NsmClient::linkDrumkit` in there in the first place is because - when under session management - Hydrogen will always be called by the NSM daemon. So, all verbose output has to be taken from it's log output (everything that goes to stderr) and there is no way to provide commandline flags for debugging. Using `NsmClient::printError` and `NsmClient::printMessage` one gets both a "[Hydrogen] " prefix required to identify the corresponding line of log and everything is visible since it's directed to stderr. On the other hand, our log messages are directed to stdout - even the ERRORLOG ones.

If either `copyPreferences` or `linkDrumkit` are required in other parts of Hydrogen too, we can put them in `Filesystem` and add some more logic for the logging. Putting them there in their current version might be more confusing then helpful

Reply:
Small correction: the original `non-session-manager` only supports messages directed to stderr. The tweaked `nsm-legacy-gui` (it's fork) of the New Session Manager does display the ERRORLOGs as well.